pkg/controller/node/nodecontroller.go

```diff
@@ -193,11 +202,31 @@ func NewNodeController(
 	return nc
 }
 
+// generateNewAvailabeCIDRs does generating new Available CIDRs index and insert
+// them into availableCIDRs set. Everytime it will generate 256 new CIDRs,
+// once there are no more CIDRs in the network, return false
+func (nc *NodeController) generateNewAvailabeCIDRs() bool {
+	if nc.firstAvailable+256 > nc.maxCIDRs {
+		glog.V(4).Infof("Out of CIDRs")
+		return false
+	}
+	for i := nc.firstAvailable; i < nc.firstAvailable+256; i++ {
+		nc.availableCIDRs.Insert(i)
+	}
+	nc.firstAvailable = nc.firstAvailable + 256
+	return true
+}
+
 // Run starts an asynchronous loop that monitors the status of cluster nodes.
 func (nc *NodeController) Run(period time.Duration) {
 	go nc.nodeController.Run(util.NeverStop)
 	go nc.podController.Run(util.NeverStop)
 	// Incorporate the results of node status pushed from kubelet to master.
+	// generate first 256 available CIDRs here
+	cidrIP := nc.clusterCIDR.IP.To4()
+	nc.firstAvailable = 0
+	nc.maxCIDRs = (256-int(cidrIP[1]))*256 - int(cidrIP[2])
+	nc.generateNewAvailabeCIDRs()
```

Comment:
This line is not needed - when they're not available `reconcileNodeCIDRs` will generate them.